Source/FieldSolver/WarpX_FDTD.H

```diff
@@ -5,7 +5,376 @@
 
 using namespace amrex;
 
-AMREX_GPU_HOST_DEVICE AMREX_INLINE
+AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+void warpx_push_bx_yee(int i, int j, int k, Array4<Real> const& Bx,
+                       Array4<Real const> const& Ey, Array4<Real const> const& Ez,
+                       Real dtsdy, Real dtsdz)
+{
+#if (AMREX_SPACEDIM == 3)
+    Bx(i,j,k) += - dtsdy * (Ez(i,j+1,k  ) - Ez(i,j,k))
+                 + dtsdz * (Ey(i,j  ,k+1) - Ey(i,j,k));
+#else
+    // Note that the 2D Cartesian and RZ versions are the same
+    Bx(i,j,0) += + dtsdz * (Ey(i,j+1,0) - Ey(i,j,0));
+#endif
+}
+
+AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+void warpx_push_by_yee(int i, int j, int k, Array4<Real> const& By,
+                       Array4<Real const> const& Ex, Array4<Real const> const& Ez,
+                       Real dtsdx, Real dtsdz)
+{
+#if (AMREX_SPACEDIM == 3)
+    By(i,j,k) += + dtsdx * (Ez(i+1,j,k  ) - Ez(i,j,k))
+                 - dtsdz * (Ex(i  ,j,k+1) - Ex(i,j,k));
+#else
+    // Note that the 2D Cartesian and RZ versions are the same
+    By(i,j,0) += + dtsdx * (Ez(i+1,j  ,0) - Ez(i,j,0))
+                 - dtsdz * (Ex(i  ,j+1,0) - Ex(i,j,0));
+#endif
+}
+
+AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+void warpx_push_bz_yee(int i, int j, int k, Array4<Real> const& Bz,
+                       Array4<Real const> const& Ex, Array4<Real const> const& Ey,
+                       Real dtsdx, Real dtsdy, Real dxinv, Real rmin)
+{
+#if (AMREX_SPACEDIM == 3)
+    Bz(i,j,k) += - dtsdx * (Ey(i+1,j  ,k) - Ey(i,j,k))
+                 + dtsdy * (Ex(i  ,j+1,k) - Ex(i,j,k));
+#else
+#ifndef WARPX_RZ
```

Comment:
I think we are trying to move away from nest preprocessor directives (in this case, `#ifndef WARPX_RZ` is inside the test on `(AMREX_SPACEDIM == 3)`).

@WeiqunZhang introduced variables `WARPX_DIM_RZ` to this effect (see commit a5df2874d). Here I think you could replace this piece of code with
```
#if   defined WARPX_DIM_3D
    Bz(i,j,k) += - dtsdx * (Ey(i+1,j  ,k) - Ey(i,j,k))
                 + dtsdy * (Ex(i  ,j+1,k) - Ex(i,j,k));
#elif defined WARPX_DIM_2D
    Bz(i,j,0) += - dtsdx * (Ey(i+1,j,0) - Ey(i,j,0));
#elif defined WARPX_DIM_RZ
    Real ru = 1. + 0.5/(rmin*dxinv + i + 0.5);
    Real rd = 1. - 0.5/(rmin*dxinv + i + 0.5);
    Bz(i,j,0) += - dtsdx * (ru*Ey(i+1,j,0) - rd*Ey(i,j,0));
#endif
```
Do you agree with this change? If so, could you include it in the rest of this PR?
@RemiLehe should we make this standard in WarpX?
Thanks!

Reply:
The changes have been implemented.